models/stdp_triplet_connection.h

```diff
@@ -0,0 +1,355 @@
+
+  /**
+   * Default Constructor.
+   * Sets default values for all parameters. Needed by GenericConnectorModel.
+   */
+  STDPTripletConnection();
+
+  /**
+   * Copy constructor.
+   * Needs to be defined properly in order for GenericConnector to work.
+   */
+  STDPTripletConnection( const STDPTripletConnection& );
+
+  /**
+   * Default Destructor.
+   */
+  ~STDPTripletConnection()
+  {
+  }
+
+  // Explicitly declare all methods inherited from the dependent base
+  // ConnectionBase. This avoids explicit name prefixes in all places
+  // these functions are used. Since ConnectionBase depends on the template
+  // parameter, they are not automatically found in the base class.
+  using ConnectionBase::get_delay_steps;
+  using ConnectionBase::get_delay;
+  using ConnectionBase::get_rport;
+  using ConnectionBase::get_target;
+
+  /**
+   * Get all properties of this connection and put them into a dictionary.
+   */
+  void get_status( DictionaryDatum& d ) const;
+
+  /**
+   * Set properties of this connection from the values given in dictionary.
+   */
+  void set_status( const DictionaryDatum& d, ConnectorModel& cm );
+
+  /**
+   * Send an event to the receiver of this connection.
+   * \param e The event to send
+   * \param t_lastspike Point in time of last spike sent.
+   * \param cp common properties of all synapses (empty).
+   */
+  void send( Event& e, thread t, double_t t_lastspike, const CommonSynapseProperties& cp );
+
+  class ConnTestDummyNode : public ConnTestDummyNodeBase
+  {
+  public:
+    // Ensure proper overriding of overloaded virtual functions.
+    // Return values from functions are ignored.
+    using ConnTestDummyNodeBase::handles_test_event;
+    port
+    handles_test_event( SpikeEvent&, rport )
+    {
+      return invalid_port_;
+    }
+  };
+
+  /*
+   * This function calls check_connection on the sender and checks if the receiver
+   * accepts the event type and receptor type requested by the sender.
+   * Node::check_connection() will either confirm the receiver port by returning
+   * true or false if the connection should be ignored.
+   * We have to override the base class' implementation, since for STDP
+   * connections we have to call register_stdp_connection on the target neuron
+   * to inform the Archiver to collect spikes for this connection.
+   *
+   * \param s The source node
+   * \param r The target node
+   * \param receptor_type The ID of the requested receptor type
+   * \param t_lastspike last spike emitted by presynaptic neuron
+   */
+  void
+  check_connection( Node& s,
+    Node& t,
+    rport receptor_type,
+    double_t t_lastspike,
+    const CommonPropertiesType& )
+  {
+    ConnTestDummyNode dummy_target;
+
+    ConnectionBase::check_connection_( dummy_target, s, t, receptor_type );
+
+    t.register_stdp_connection( t_lastspike - get_delay() );
+  }
+
+  void
+  set_weight( double_t w )
+  {
+    weight_ = w;
+  }
+
+private:
+  inline double_t // TBD
```

Comment:
Was the `// TBD` leftover from earlier or is it actually still hinting at something to be done?